alonzo/formal-spec/properties.tex

```diff
@@ -140,18 +164,75 @@ \section{Formal Properties}
 \begin{property}[Paying fees]
   \label{prop:pay-fees}
 
-A transaction always pays into the fee pot.
+If the collateral percent value is set to at least $100$, every transaction pays
+at least the minimum transaction fee amount into the fee pot.
 
 \begin{lemma}
   For all environments $e$, transactions $tx$ and states $s, s'$ such that
   \begin{equation*}
-    e\vdash s\trans{ledger}{tx}s',
+    e\vdash s\trans{ledger}{tx}s'~\wedge~\fun{collateralPercent}~(\field_{pp}~(s))~\geq~100
```

Comment:
i think the usefulness of this lemma comes from the idea that we want to establish some conditions under which we can guarantee that the transaction pays the minfee

Reply:
I'm not sure we care about the minfee specifically, only that it pays a fee if we want it to. But the statement using `φ` is more general, and so this can be a corollary

Reply:
i'll try this

Reply:
If the collateralPercent is smaller than 100, we dont have anything to bound the fee with. I dont think it is useful to say anything along the lines of "depending on collateralPercent, and assuming one of the fee params is > 0,
`0 < totalCollateralBalance < minfee <= txfee` OR
`0 < minfee < totalCollateralBalance` .... etc. and consider every other case, then for fee params = 0 case too

I think this property is clear and useful for testing as is